From db48271175c0566bb5e59776a23297d64aec9fcd Mon Sep 17 00:00:00 2001
From: Alison Wang <b18965@freescale.com>
Date: Thu, 21 Jul 2016 18:13:29 +0800
Subject: [PATCH 83/93] armv8: fsl-layerscape: Fix the conflict between PPA
 and AArch32 support

As there is conflict between PPA and AArch32 support, this patch is to
add a new defconfig for AArch32 support and move the definition for PPA
to the board defconfig.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
---
 configs/ls1012afrdm_aarch32_qspi_defconfig    |   11 +++++++++++
 configs/ls1012afrdm_qspi_defconfig            |    1 +
 configs/ls1012ardb_aarch32_qspi_defconfig     |   11 +++++++++++
 configs/ls1012ardb_qspi_SECURE_BOOT_defconfig |    1 +
 configs/ls1012ardb_qspi_defconfig             |    1 +
 include/configs/ls1012afrdm.h                 |    1 -
 include/configs/ls1012ardb.h                  |    1 -
 7 files changed, 25 insertions(+), 2 deletions(-)
 create mode 100644 configs/ls1012afrdm_aarch32_qspi_defconfig
 create mode 100644 configs/ls1012ardb_aarch32_qspi_defconfig

diff --git a/configs/ls1012afrdm_aarch32_qspi_defconfig b/configs/ls1012afrdm_aarch32_qspi_defconfig
new file mode 100644
index 0000000..1bb9080
--- /dev/null
+++ b/configs/ls1012afrdm_aarch32_qspi_defconfig
@@ -0,0 +1,11 @@
+CONFIG_ARM=y
+CONFIG_TARGET_LS1012AFRDM=y
+CONFIG_SYS_EXTRA_OPTIONS="QSPI_BOOT"
+# CONFIG_CMD_IMLS is not set
+# CONFIG_FSL_LS_PPA is not set
+CONFIG_SYS_NS16550=y
+CONFIG_DEFAULT_DEVICE_TREE="fsl-ls1012a-frdm"
+CONFIG_OF_CONTROL=y
+CONFIG_DM=y
+CONFIG_SPI_FLASH=y
+CONFIG_DM_SPI=y
diff --git a/configs/ls1012afrdm_qspi_defconfig b/configs/ls1012afrdm_qspi_defconfig
index e27181c..1f69db7 100644
--- a/configs/ls1012afrdm_qspi_defconfig
+++ b/configs/ls1012afrdm_qspi_defconfig
@@ -2,6 +2,7 @@ CONFIG_ARM=y
 CONFIG_TARGET_LS1012AFRDM=y
 CONFIG_SYS_EXTRA_OPTIONS="QSPI_BOOT"
 # CONFIG_CMD_IMLS is not set
+CONFIG_FSL_LS_PPA=y
 CONFIG_SYS_NS16550=y
 CONFIG_DEFAULT_DEVICE_TREE="fsl-ls1012a-frdm"
 CONFIG_OF_CONTROL=y
diff --git a/configs/ls1012ardb_aarch32_qspi_defconfig b/configs/ls1012ardb_aarch32_qspi_defconfig
new file mode 100644
index 0000000..df16f3b
--- /dev/null
+++ b/configs/ls1012ardb_aarch32_qspi_defconfig
@@ -0,0 +1,11 @@
+CONFIG_ARM=y
+CONFIG_TARGET_LS1012ARDB=y
+CONFIG_SYS_EXTRA_OPTIONS="QSPI_BOOT"
+# CONFIG_CMD_IMLS is not set
+# CONFIG_FSL_LS_PPA is not set
+CONFIG_SYS_NS16550=y
+CONFIG_DEFAULT_DEVICE_TREE="fsl-ls1012a-rdb"
+CONFIG_OF_CONTROL=y
+CONFIG_DM=y
+CONFIG_SPI_FLASH=y
+CONFIG_DM_SPI=y
diff --git a/configs/ls1012ardb_qspi_SECURE_BOOT_defconfig b/configs/ls1012ardb_qspi_SECURE_BOOT_defconfig
index 92a95a8..4091503 100644
--- a/configs/ls1012ardb_qspi_SECURE_BOOT_defconfig
+++ b/configs/ls1012ardb_qspi_SECURE_BOOT_defconfig
@@ -2,6 +2,7 @@ CONFIG_ARM=y
 CONFIG_TARGET_LS1012ARDB=y
 CONFIG_SYS_EXTRA_OPTIONS="QSPI_BOOT,SECURE_BOOT"
 # CONFIG_CMD_IMLS is not set
+# CONFIG_FSL_LS_PPA is not set
 CONFIG_SYS_NS16550=y
 CONFIG_DEFAULT_DEVICE_TREE="fsl-ls1012a-rdb"
 CONFIG_OF_CONTROL=y
diff --git a/configs/ls1012ardb_qspi_defconfig b/configs/ls1012ardb_qspi_defconfig
index f819038..e025247 100644
--- a/configs/ls1012ardb_qspi_defconfig
+++ b/configs/ls1012ardb_qspi_defconfig
@@ -2,6 +2,7 @@ CONFIG_ARM=y
 CONFIG_TARGET_LS1012ARDB=y
 CONFIG_SYS_EXTRA_OPTIONS="QSPI_BOOT"
 # CONFIG_CMD_IMLS is not set
+CONFIG_FSL_LS_PPA=y
 CONFIG_SYS_NS16550=y
 CONFIG_DEFAULT_DEVICE_TREE="fsl-ls1012a-rdb"
 CONFIG_OF_CONTROL=y
diff --git a/include/configs/ls1012afrdm.h b/include/configs/ls1012afrdm.h
index 0e35fe9..a736687 100644
--- a/include/configs/ls1012afrdm.h
+++ b/include/configs/ls1012afrdm.h
@@ -10,7 +10,6 @@
 #include "ls1012a_common.h"
 
 #ifndef CONFIG_SECURE_BOOT
-#define CONFIG_FSL_LS_PPA
 #if defined(CONFIG_FSL_LS_PPA)
 #define CONFIG_ARMV8_PSCI
 #define CONFIG_SYS_LS_PPA_DRAM_BLOCK_MIN_SIZE	(1UL * 1024 * 1024)
diff --git a/include/configs/ls1012ardb.h b/include/configs/ls1012ardb.h
index d78edd2..400fa61 100644
--- a/include/configs/ls1012ardb.h
+++ b/include/configs/ls1012ardb.h
@@ -9,7 +9,6 @@
 
 #include "ls1012a_common.h"
 
-#define CONFIG_FSL_LS_PPA
 #if defined(CONFIG_FSL_LS_PPA)
 #define CONFIG_ARMV8_PSCI
 #define CONFIG_SYS_LS_PPA_DRAM_BLOCK_MIN_SIZE		(1UL * 1024 * 1024)
-- 
1.7.9.5

